### PR TITLE
Add Ability To Handle Zero to Many Assessments in a Given Yukon Grant Approval Letter

### PR DESCRIPTION
**Pull request by klondikemarlen**
# Context

A given funding request letter may have zero to many assessments.
The Yukon Grant Student Approval and Yukon Grant Institution Approval need the ability to support a Funding Request that has no assessments, or possibly many assessment.

See the following files:
- `src/api/serializers/funding-requests/letters/yukon-grant-student-approval-template-serializer.ts`
- `src/api/serializers/funding-requests/letters/yukon-grant-institution-approval-template-serializer.ts`
- `src/api/templates/admin/application-letter/approval/yukon-grant-student.handlebars`
- `src/api/templates/admin/application-letter/approval/yukon-grant-institution.handlebars`
- ... others?

Also investigate possibility of zero disbursements and what that would look like in the template.
Also add some diagramming of the basic data flow of letter creation.

## Questions
- [x] How should I separate/distinguish assessment sections? Should I give assessments a title like "assessment 1" ...2 ...3? Or something else?

# Screenshots

Example of Yukon Grant Student Approval Letter without any assessments or disbursements
image

Example of letter with a two disbursements and one assessments.
image

Example of letter with two disbursements and two assessments.
image

Example diagram of funding request letter template complexity
Diagram, Funding Request Letters, 2023-08-24

# Testing Instructions

1. Boot the api and database services via `dev up`
2. Boot the web service via `cd src/web && npm run start`
3. Log in to the app via the UI at http://localhost:8080
4. Go to http://localhost:3000/api/v2/admin/funding-requests/266/letters/student.json and check that you can now serialize a letter without assessments or disbursements.
5. Go to http://localhost:3000/api/v2/admin/funding-requests/266/letters/student.html to inspect the html preview
6. Go to http://localhost:3000/api/v2/admin/funding-requests/266/letters/student.pdf to check the final pdf of the letter.
7. Go to http://localhost:3000/api/v2/admin/funding-requests/69473/letters/student.pdf for a letter with assessments.
8. Open your database editor go to the `assessment` table. Duplicated the row with `funding_request_id = 69473`
9. Go to http://localhost:3000/api/v2/admin/funding-requests/69473/letters/student.pdf and check that you now see current/previous headers around the assessment data.

### CODE REVIEW

**Comment by datajohnson** [comment]
Question: `How should I separate/distinguish assessment sections? Should I give assessments a title like "assessment 1" ...2 ...3? Or something else?`

Answer: `Current shows first and just lump all the others below under the section heading of: Previous Approval(s) for this academic year'`

image

**Comment by klondikemarlen** [comment]
Where do the fields for the names of the assessments come from? e.g 
image

What field/table is `Canada Student Financial Assistance Full-Time` pulled from?

File: src/api/templates/admin/application-letter/approval/yukon-grant-student.handlebars [hunk 1]
```diff
@@ -46,39 +46,44 @@
...
+      <span>
+        <strong style="inline-block w-[150px]">Approv. Weeks:</strong>
+        <span>{{this.approvalWeeks}}</span>
+      </span>
+      <br />
+      <span>
+        <strong style="inline-block w-[150px]">Travel/Airfare:</strong>
+        <span>{{centsToDollarString this.travelAndAirFairCostInCents}}</span>
+      </span>
+    {{/each}}
```
Comment (klondikemarlen) [inline src/api/templates/admin/application-letter/approval/yukon-grant-student.handlebars:66]: @datajohnson This is the current code, I'm not sure what fields to pull the new logic from.

Comment (klondikemarlen) [reply src/api/templates/admin/application-letter/approval/yukon-grant-student.handlebars]: Resolving, letter example is for other letter and this one is less complex.

**Review by datajohnson — APPROVED** [review]

**Pull request merged**